### DISCUSSION TITLE
Show code diffs on composer update

### DISCUSSION THREAD

**Discussion by d3vpunk** [composer/composer#10109] [General]
First of all, composer is the best thing what happened to the PHP community. Great tool.

But with all dependency managers there is always the problem that foreign code is loaded.
Basically everything is based on trust. As a project owner who add a dependency, i trust the package owner "A". Then again, for the dependencies of that dependency the package owner "A" needs to trust the package owner "B" and so on.
This makes it hard for the project owner to keep control over code that become part of your project.

Idea
----- 
Is it an idea to make it possible to show the code diffs on a changed package on composer update?
For example,  a new minor of "symfony/process" is incoming - and at that point let the user choice with a prompt to show the diffs of that minor version. Of course, the "symfony/process" is trusty - but there are less know packages where you want to review the changes. And reviewing the changes manually is somewhat cumbersome.. 1) google github package, 2) selecting good versions to compare... By the way - a direct link to the compare page in my CLI is also be an idea.

This could be a nice addition to the PHP world, because that way there are potentially MANY more eyes to see code changes and in turn this increases code quality and security.

That just a brainstorm. I'm curious what you people are thinking ;)

**Comment by stof** [comment]
See https://github.com/davidrjonas/composer-lock-diff. It does that kind of things (well, it gives you a link to see the diff on github rather than showing it directly)

**Comment by d3vpunk** [comment]
Oh that is perfect. 
But isn't it an idea to integrate that idea into composer itself somehow? That way many more developers potentially see it. Because now it's an additional (great) library needs actively be downloaded and activated - also additional commands are needed to see that information

There is visually much space to show what additional information like that links:
![composer](https://user-images.githubusercontent.com/1311487/133584155-92110dda-6d39-4af2-a7a8-fab66710c15b.png)

**Reply by stof** [reply]
> There is visually much space to show what additional information like that links:

Not all consoles are that wide.

**Reply by d3vpunk** [reply]
That's surely true. But that metainformation could be conditionally be shown based on the width of the CLI.
